dashboard/app/jobs/evaluate_rubric_job.rb

```diff
@@ -30,15 +23,83 @@ class EvaluateRubricJob < ApplicationJob
     }
   }
 
-  def perform(user_id:, script_level_id:)
+  # Ensure that the RubricAiEvaluation exists as an argument to the job
+  private def pass_in_or_create_rubric_ai_evaluation(job)
+    # Get the first argument to perform() which is the hash of named arguments
+    options = job.arguments.first
+
+    # Get the level containing the rubric
+    script_level = ScriptLevel.find(options[:script_level_id])
+
+    rubric = Rubric.find_by!(lesson_id: script_level.lesson.id, level_id: script_level.level.id)
+    return nil unless rubric
+
+    user = User.find(options[:user_id])
+    channel_id = get_channel_id(user, script_level)
+    _owner_id, project_id = storage_decrypt_channel_id(channel_id)
+
+    # Create a queued record of this work request (if none were given)
+    rubric_ai_evaluation_id = options[:rubric_ai_evaluation_id]
+    rubric_ai_evaluation = (rubric_ai_evaluation_id && RubricAiEvaluation.find(rubric_ai_evaluation_id)) || RubricAiEvaluation.create!(
+      user_id: options[:user_id],
+      requester_id: options[:requester_id],
+      rubric: Rubric.find_by!(lesson_id: script_level.lesson.id, level_id: script_level.level.id),
+      status: SharedConstants::RUBRIC_AI_EVALUATION_STATUS[:QUEUED],
+      project_id: project_id,
+    )
+
+    # Set it back so the job can reference it
+    options[:rubric_ai_evaluation_id] = rubric_ai_evaluation.id
+
+    # Return the Rubric
+    rubric_ai_evaluation
+  end
+
+  before_enqueue do |job|
+    rubric_ai_evaluation = pass_in_or_create_rubric_ai_evaluation(job)
+    if rubric_ai_evaluation
```

Comment:
this seems like a given. what's the situation where we wouldn't have a rubric_ai_evaluation at this point?

Reply:
There isn't one! All cases where it doesn't produce that record would be cases it would raise an exception anyway. Hm... the reminds me that I need to catch those exceptions in the `rescue_from` so I don't shadow the original exception.